src/frontend/components/UI/QuestsViewer/index.tsx

```diff
@@ -0,0 +1,162 @@
+import useGetQuest from 'frontend/hooks/useGetQuest'
+import useGetSteamGame from 'frontend/hooks/useGetSteamGame'
+import useAuthSession from 'frontend/hooks/useAuthSession'
+import { useTranslation } from 'react-i18next'
+
+export interface QuestsViewerProps {
+  projectId: string
+}
+
+export function QuestsViewer({ projectId: appName }: QuestsViewerProps) {
+  const questsResults = useGetQuests(appName)
+  const quests = questsResults?.data?.data
+  const { t } = useTranslation()
+
+  const [selectedQuestId, setSelectedQuestId] = useState<number | null>(null)
+  const questResult = useGetQuest(selectedQuestId)
+  let questLog = null
+  if (Array.isArray(quests)) {
+    const questsUi = quests.map((val) => {
+      const questUi_i: QuestLogInfo = {
+        questType: 'REPUTATION',
+        title: val.name,
+        state: 'ACTIVE',
+        onClick: () => {
+          // deselect
+          if (selectedQuestId === val.id) {
+            setSelectedQuestId(null)
+            return
+          }
+          //select
+          setSelectedQuestId(val.id)
+        },
+        selected: selectedQuestId === val.id,
+        id: val.id
+      }
+      return questUi_i
+    })
+    questLog = <QuestLog quests={questsUi} className={styles.questLog} />
+  } else if (questsResults?.data.isLoading || questsResults?.data.isFetching) {
+    questLog = (
+      <QuestLog quests={[]} className={styles.questLog} loading={true} />
+    )
+  }
+
+  let questDetails = null
+  const questMeta = questResult.data.data as Quest
+
+  const getSteamGameResult = useGetSteamGame(
+    questMeta?.eligibility?.steam_games ?? []
+  )
+
+  const steamGames: Game[] =
+    getSteamGameResult?.data?.map((val, index) => ({
+      /* eslint-disable-next-line */
+      // @ts-ignore
+      title: val.data?.name ?? index,
+      /* eslint-disable-next-line */
+      // @ts-ignore
+      imageUrl: val.data?.capsule_image ?? '',
+      loading: val.isLoading || val.isFetching
+    })) ?? []
+
+  const [collapseIsOpen, setCollapseIsOpen] = useState(false)
+  const session = useAuthSession()
+  const accounts = session.data?.linkedAccounts
+  const steamIsLinked = accounts?.has('steam')
+  const i18n = {
+    reward: t('quest.reward', 'Reward'),
+    associatedGames: t('quest.associatedGames', 'Associated games'),
+    linkSteamAccount: t(
+      'quest.linkAccount',
+      'Link your Steam account to check eligibility.'
+    ),
+    needMoreAchievements: t(
+      'quest.needMoreAchievements',
+      `You need to have completed {{percent}}% of the achievements in one of these games.`,
+      { percent: questMeta?.eligibility?.completion_threshold ?? '??' }
+    ),
+    claim: steamIsLinked
+      ? t('quest.claimAll', 'Claim all')
+      : t('quest.connectSteamAccount', 'Connect Steam account'),
+    questType: {
+      REPUTATION: t('quest.reputation', 'Reputation')
+    }
+  }
+
+  if (
+    selectedQuestId !== null &&
+    Object.hasOwn(questResult?.data?.data ?? {}, 'name')
```

Comment:
this looks off, why do you need to do this? can typing the query help remove this?

Reply:
Yeah. Just removed it. I was having issues with the keys/re-rendering at one point and wanted to ensure that each component had a unique key. was using this to debug

This http response is really an unknown type though. maybe we can add zod to the ipcmain handler at some point